pkg/daemon/daemon.go

```diff
@@ -272,6 +282,15 @@ func (dn *Daemon) applyNodePTPProfiles() error {
 	glog.Infof("updating NodePTPProfiles to:")
 	runID := 0
 	slices.SortFunc(dn.ptpUpdate.NodeProfiles, func(a, b ptpv1.PtpProfile) int {
+		aHasPhc2sysOpts := a.Phc2sysOpts != nil && *a.Phc2sysOpts != ""
+		bHasPhc2sysOpts := b.Phc2sysOpts != nil && *b.Phc2sysOpts != ""
+		//sorted in ascending order
+		// here having phc2sysOptions is considered a low number
+		if !aHasPhc2sysOpts && bHasPhc2sysOpts {
+			return -1 //  a>b return -1
```

Comment:
This is not correct, per https://pkg.go.dev/slices#SortFunc: 
cmp(a, b) should return a negative number when a < b, a positive number when a > b and zero when a == b.

Reply:
It is custom comparison.